src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -313,6 +333,8 @@ def _save_cloud(cloud, overwrite=False):
     except configparser.DuplicateSectionError:
         if not overwrite:
             raise CloudAlreadyRegisteredException(cloud.name)
+    if cloud.profile is not None:
```

Comment:
do we need to differentiate between `None` and empty string? If yes, what does empty string imply?